### PR TITLE
Support eval code debugging

### PR DESCRIPTION
**Pull request by zherczeg**

### CODE REVIEW

**Comment by clover2123** [comment]
Would you please add more test cases?
For example, such as local eval call and global eval call (indirect eval call) inside a function

File: src/interpreter/ByteCodeGenerator.cpp [hunk 1]
```diff
@@ -269,7 +260,7 @@ void ByteCodeGenerator::collectByteCodeLOCData(Context* context, InterpretedCode
 
     // Parsing
     Node* ast = nullptr;
-    if (codeBlock->isGlobalScope()) {
+    if (codeBlock->isGlobalScope() || codeBlock->isEvalCode()) {
```
Comment (zherczeg) [inline src/interpreter/ByteCodeGenerator.cpp:263]: Without this, backtrace in `eval("g = 2\ng = 3")` enters an infinite loop. I don't know if this fix is correct.

Comment (clover2123) [reply src/interpreter/ByteCodeGenerator.cpp:263]: We need to parse the entire source code for global code or eval code.
So, please fix this line as below
```c++
if (codeBlock->isGlobalCodeBlock()() || codeBlock->isEvalCode()) {
```

Comment (zherczeg) [reply src/interpreter/ByteCodeGenerator.cpp]: Thank you, updated.

**Review by clover2123 — APPROVED** [review]
LGTM

**Review by ksh8281 — APPROVED** [review]

**Pull request merged**